pages/content/amp-dev/documentation/guides-and-tutorials/optimize-measure/signed-exchange.md

```diff
@@ -32,42 +32,43 @@ To implement Signed Exchanges, you must meet the following requirements:
 
+In order to generate the certificate, the Certificate Authority (CA) will require a Certificate Signing Request (CSR), which can be generated by `openssl`. An example CSR for `ampbyexample.com`:
 
-```html
+```sh
+# generate private key (if necessary)
 $ openssl ecparam -out ampbyexample-packager.key -name prime256v1 -genkey
-
+# generate CSR (the file ampbyexample-packager.csr)
 $ openssl req -new -key ampbyexample-packager.key -nodes -out ampbyexample-packager.csr -subj "/C=US/ST=California/L=Mountain View/O=Google LLC/CN=ampbyexample.com"
 ```
+## Determine which URLs will be signed 
+
+You will need to create a URL pattern that defines which documents should be signed. It is critical that private content, such as personalized information should not be signed, to avoid sending misleading or incorrect content.
 
-### Additional certificates 
-The [`amppackager`](https://github.com/ampproject/amppackager) needs two files for encryption: a **CertFile** and a **KeyFile**. These are specified in the `amppkg.toml` config file. 
+For performance purposes, the packager should only be passed valid AMP documents as input. Some invalid AMP documents are fine if needed, but you should avoid sending all traffic through the packager. 
 
-The **KeyFile** is the private key. This may have been generated via an openssl command such as:
+## Deploy packager to a staging server
 
+You should set Signed Exchanges up on a staging server to verify your setup correct before migrating to production.
 
-```html
-openssl ecparam -out ampdev-packager.key -name prime256v1 -genkey
-```
+We recommend using [`amppackager`](https://github.com/ampproject/amppackager/blob/master/README.md) to generate signed exchanges. However if this is not a good fit for your production environment you can instead use the command-line clients [`transform`](https://github.com/ampproject/amppackager/blob/master/transformer/README.md) and [`gen-signedexchange`](https://github.com/WICG/webpackage/tree/master/go/signedexchange), and handle the content negotiation and certificate management tasks yourself.
 
+The following instructions apply to the deployments using `amppackager`.
 
-The private key is used to _generate_ the CSR which is sent to the certificate authority, but is otherwise kept private.
+### Configuration
 
-If it is generated via the openssl command above, it will be in `PEM` format. It should look something like the following:
+[`amppackager`](https://github.com/ampproject/amppackager)'s config file (`amppkg.toml`) calls for a **CertFile** and a **KeyFile**.
 
+The **KeyFile** is the private key (`ampbyexample-packager.key` in the example above). It should look like this:
```

Comment:
Maybe add a disclaimer that this is a sample key file, and people should not share their real keys.

Reply:
Done!